sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -383,7 +385,14 @@ case class AlterTableRenamePartitionCommand(
   extends RunnableCommand {
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    sparkSession.sessionState.catalog.renamePartitions(
```

Comment:
We got the following errors:
```
Partition not found in table 'tab1' database 'default':
ds -> 2008-04-08
hr -> 11;
org.apache.spark.sql.catalyst.analysis.NoSuchPartitionException: Partition not found in table 'tab1' database 'default':
ds -> 2008-04-08
hr -> 11;
	at org.apache.spark.sql.hive.client.HiveClientImpl$$anonfun$renamePartitions$1$$anonfun$apply$mcV$sp$5$$anonfun$16.apply(HiveClientImpl.scala:502)
	at org.apache.spark.sql.hive.client.HiveClientImpl$$anonfun$renamePartitions$1$$anonfun$apply$mcV$sp$5$$anonfun$16.apply(HiveClientImpl.scala:502)
```

Basically, Hive does not recognize the partitioning info of data source tables.